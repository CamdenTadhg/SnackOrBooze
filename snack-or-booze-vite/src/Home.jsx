import React from "react";
import {Link} from 'react-router-dom';
import { Card, CardBody, CardTitle, ListGroup, ListGroupItem} from "reactstrap";
import 'bootstrap/dist/css/bootstrap.min.css';

//renders home page of app
function Home({snacks, drinks}) {
  return (
    <section className="col-md-8">
      <Card>
        <CardBody className="text-center">
          <CardTitle>
            <h5 className="font-weight-bold">
              Welcome to Silicon Valley's premier dive cafe!
            </h5>
          </CardTitle>
            <ListGroup>
              <Link to={'/snacks'}>
                <ListGroupItem>Snacks: {snacks.length}</ListGroupItem>
              </Link>
              <Link to={'/drinks'}>
                <ListGroupItem>Drinks: {drinks.length}</ListGroupItem>
              </Link>
            </ListGroup>
        </CardBody>
      </Card>
    </section>
  );
}

export default Home;