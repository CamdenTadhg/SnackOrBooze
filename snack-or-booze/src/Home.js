import React from "react";
import {Link} from 'react-router-dom';
import { Card, CardBody, CardTitle, CardText, ListGroup, ListGroupItem} from "reactstrap";

function Home({snacks, drinks}) {
  return (
    <section className="col-md-8">
      <Card>
        <CardBody className="text-center">
          <CardTitle>
              Welcome to Silicon Valley's premier dive cafe!
          </CardTitle>
            <ListGroup>
              <Link to={'/snacks'}>
                <ListGroupItem>Snacks: {snacks.length}</ListGroupItem>
              </Link>
              <Link to={'/drinks'}>
                <ListGroupItem>Drinks: {drinks.length}</ListGroupItem>
              </Link>
            </ListGroup>
        </CardBody>
      </Card>
    </section>
  );
}

export default Home;
