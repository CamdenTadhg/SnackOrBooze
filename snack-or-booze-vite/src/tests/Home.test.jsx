import {render} from '@testing-library/react';
import {test, expect} from 'vitest';
import Home from '../Home';
import {MemoryRouter} from 'react-router-dom';

const testSnacks = ['nachos', 'hummus', 'arugula-and-walnut-salad', 'cookie', 'chipscheese'];
const testDrinks = ['martini', 'negroni', 'gin-and-tonic', 'diet-coke', 'kir', 'manhattan', 'mary'];

test('renders the Home component', () => {
    render(
        <MemoryRouter>
            <Home snacks={testSnacks} drinks={testDrinks}/>
        </MemoryRouter>)
});

test('it matches snapshot', async () => {
    const home = render(
        <MemoryRouter>
            <Home snacks={testSnacks} drinks={testDrinks}/>
        </MemoryRouter>);
    expect(home).toMatchSnapshot();
});

test('renders the right content', () => {
    const {getByText} = render(
        <MemoryRouter>
            <Home snacks={testSnacks} drinks={testDrinks}/>
        </MemoryRouter>);
    expect(getByText('Snacks: 5', {exact: false})).toBeInTheDocument();
    expect(getByText('Drinks: 7', {exact: false})).toBeInTheDocument();
})