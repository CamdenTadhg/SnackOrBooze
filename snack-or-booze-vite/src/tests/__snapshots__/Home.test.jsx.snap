// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="col-md-8"
      >
        <div
          class="card"
        >
          <div
            class="text-center card-body"
          >
            <div
              class="card-title"
            >
              <h5
                class="font-weight-bold"
              >
                Welcome to Silicon Valley's premier dive cafe!
              </h5>
            </div>
            <ul
              class="list-group"
            >
              <a
                href="/snacks"
              >
                <li
                  class="list-group-item"
                >
                  Snacks: 
                  5
                </li>
              </a>
              <a
                href="/drinks"
              >
                <li
                  class="list-group-item"
                >
                  Drinks: 
                  7
                </li>
              </a>
            </ul>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="col-md-8"
    >
      <div
        class="card"
      >
        <div
          class="text-center card-body"
        >
          <div
            class="card-title"
          >
            <h5
              class="font-weight-bold"
            >
              Welcome to Silicon Valley's premier dive cafe!
            </h5>
          </div>
          <ul
            class="list-group"
          >
            <a
              href="/snacks"
            >
              <li
                class="list-group-item"
              >
                Snacks: 
                5
              </li>
            </a>
            <a
              href="/drinks"
            >
              <li
                class="list-group-item"
              >
                Drinks: 
                7
              </li>
            </a>
          </ul>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
